clients/filesystem-hadoop3/src/test/java/org/apache/gravitino/filesystem/hadoop/TestGvfsBase.java

```diff
@@ -607,4 +651,52 @@ public void testGetDefaultBlockSize() throws IOException {
       assertEquals(32 * 1024 * 1024, fs.getDefaultBlockSize(managedFilesetPath));
     }
   }
+
+  @Test
+  public void testConvertFileStatusPathPrefix() throws IOException {
+    try (GravitinoVirtualFileSystem fs =
+        (GravitinoVirtualFileSystem) managedFilesetPath.getFileSystem(conf)) {
+      FileStatus fileStatus =
+          new FileStatus(1024, false, 1, 32 * 1024 * 1024, 1024, new Path("hdfs://hive:9000/test"));
+      // storage location end with "/"
+      String storageLocation = "hdfs://hive:9000/";
+      String virtualLocation = "gvfs://fileset/test_catalog/tmp/test_fileset";
```

Comment:
Do we support virtual path like this "gvfs://fileset/test_catalog/tmp/test_fileset/"? @xloya

Reply:
Yeah, we support this case. Because the virtual location we concat is in a format similar to "gvfs://fileset/catalog/tmp/test_fileset", no matter whether the virtual path actually accessed has a trailing slash or not, it can be replaced correctly as long as the storage location is handled correctly. Actually when we use `new Path("gvfs://fileset/test_catalog/tmp/test_fileset/")`, the tailing slash will be truncated by Hadoop path class.

Reply:
> the tailing slash will be truncated by Hadoop path class.

What if for the python code, do we have similar guarantees?

Reply:
Yes, the current logic does not rely on the external truncate logic, and python can handle it correctly too.